sopel/tools/memories.py

```diff
@@ -161,3 +230,57 @@ def __contains__(self, key):
 
     def __setitem__(self, key: Optional[str], value):
         super().__setitem__(self._make_key(key), value)
+
+    def setdefault(self, key: str, default=None):
+        return super().setdefault(self._make_key(key), default)
+
+    def __delitem__(self, key: str):
+        super().__delitem__(self._make_key(key))
+
+    def copy(self):
+        return type(self)(self, identifier_factory=self.make_identifier)
+
+    def get(self, key: str, default=_NO_DEFAULT):
+        if default is _NO_DEFAULT:
+            return super().get(self._make_key(key))
+        return super().get(self._make_key(key), default)
+
+    def pop(self, key: str, default=_NO_DEFAULT):
+        if default is _NO_DEFAULT:
+            return super().pop(self._make_key(key))
+        return super().pop(self._make_key(key), default)
+
+    def update(self, maybe_mapping=tuple()):
+        super().update(self._convert_keys(maybe_mapping))
+
+    def __or__(self, other):
+        if not isinstance(other, dict):
+            return NotImplemented
+
+        # self on the left, so other's keys overwrite
+        new = self.copy()
+        new.update(other)
+        return new
+
+    def __ror__(self, other):
+        if not isinstance(other, dict):
+            return NotImplemented
+
+        # self on the right, so keep only new keys from other
+        new = self.copy()
+        new.update((k, v) for k, v in other.items() if k not in self)
+        return new
+
+    def __ior__(self, other):
+        if not isinstance(other, dict):
+            return NotImplemented
+        self.update(other)
+        return self
+
+    def __eq__(self, other):
+        if not isinstance(other, dict):
+            return NotImplemented
+        return super().__eq__(other)
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
```

Comment:
Nit: This _could_ be omitted, what's written here [is the default behavior](https://docs.python.org/3/reference/datamodel.html#object.__ne__):

> For `__ne__()`, by default it delegates to `__eq__()` and inverts the result unless it is NotImplemented.

```suggestion
```

Reply:
Since we're nitpicking, that's the default behavior for `object.__ne__()` and not necessarily `dict.__ne__()`, which is what we'd get if this explicit (is better than implicit) override didn't exist.

A better question is, does this behave correctly `if not isinstance(other, dict)`? Even if the override here is removed? (The answer is no. Need to fix that and add test cases for it.)

Reply:
Aha, I stand corrected!